apps/studio.giselles.ai/app/(main)/tasks/[taskId]/ui/experimental/task-data.ts

```diff
@@ -153,11 +211,7 @@ async function getTaskInput(taskId: TaskId) {
 
 export async function getTaskData(taskId: TaskId): Promise<UITask> {
 	const task = await giselle.getTask({ taskId });
-	if (task.nodeIdsConnectedToEnd === undefined) {
-		// This page expects a "new" task shape that includes nodeIdsConnectedToEnd.
-		// If it's missing, fail fast to surface data inconsistencies early.
-		throw new Error(`Task ${taskId} is missing nodeIdsConnectedToEnd`);
-	}
+	const nodeIdsConnectedToEnd = task.nodeIdsConnectedToEnd;
```

Comment:
Please tell me why this change is necessary. The tasks displayed here are tasks via App, and App always has EndNode and LowTrigger.